api/src/controllers/websocket_rpc.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+module.exports = WebsocketRpcControllerFactory
+
+const ws = require('ws')
+const debug = require('debug')('ilp-kit:websocket-rpc')
+const Connector = require('../lib/connector')
+
+function WebsocketRpcControllerFactory (deps) {
+  const connector = deps(Connector)
+
+  return class WebsocketRpcController {
+    static init (server) {
+     const socketServer = new ws.Server({
+        perMessageDeflate: false,
+        server
+      })
+
+      debug('listening for websocket connections')
+      socketServer.on('connection', socket => {
+        const url = socket.upgradeReq.url
+        const params = url.match(/^\/api\/peers\/rpc\/ws\/(.+?)\/(.+?)\/?$/)
+        if (!params) {
+          debug('Error: no ws at "' + url + '"')
+          socket.close()
+          return
+        }
+
+        const [ , prefix, token ] = params
+        let plugin
+
+        try {
+          plugin = connector.getPlugin(prefix)
```

Comment:
I know this is how the current plugin-virtual does it, but maybe now is a good time to improve that situation?
Add this point we can just instantiate the plugin instance and pass the socket in the config, instead of requiring that the plugin instance already exists and then having to call addSocket.

Reply:
The plugin is only useful if it's part of the connector, though. Otherwise it wouldn't have any routes